### PR TITLE
Remove global cfg variable (fixes #2294)

### LINKED ISSUE #2294: syncthing -upgrade is teh bork
```
$ syncthing -upgrade
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0xefb28]

goroutine 1 [running]:
github.com/syncthing/syncthing/lib/config.(*Wrapper).Options(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/syncthing/syncthing/lib/config/wrapper.go:255 +0xd8
main.main()
	/go/src/github.com/syncthing/syncthing/cmd/syncthing/main.go:348 +0x2551

goroutine 5 [syscall]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/usr/local/go/src/os/signal/signal_unix.go:27 +0x35

goroutine 7 [chan receive]:
main.trackCPUUsage()
	/go/src/github.com/syncthing/syncthing/cmd/syncthing/gui_unix.go:24 +0xec
created by main.init·3
	/go/src/github.com/syncthing/syncthing/cmd/syncthing/gui_unix.go:17 +0x25
```

### PR DESCRIPTION
**Pull request by calmh**
Not necessarily the easiest way to fix just this bug, but the root cause
was using the (at that point uninitialized) cfg variable, so it seemed
sensible to just get rid of it to avoid that kind of crap.

### CODE REVIEW

File: cmd/syncthing/gui.go [hunk 1]
```diff
@@ -197,18 +197,18 @@ func (s *apiSvc) Serve() {
...
 	// Wrap everything in basic auth, if user/password is set.
-	if len(s.cfg.User) > 0 && len(s.cfg.Password) > 0 {
-		handler = basicAuthAndSessionMiddleware("sessionid-"+s.id.String()[:5], s.cfg, handler)
+	if len(s.cfg.GUI().User) > 0 && len(s.cfg.GUI().Password) > 0 {
+		handler = basicAuthAndSessionMiddleware("sessionid-"+s.id.String()[:5], s.cfg.GUI(), handler)
 	}
 
 	// Redirect to HTTPS if we are supposed to
-	if s.cfg.UseTLS {
+	if s.cfg.GUI().UseTLS {
```
Comment (AudriusButkevicius) [inline cmd/syncthing/gui.go]: This whole section above is calling GUI() excessively. Store a copy in a var and be done with it.

Comment (calmh) [inline cmd/syncthing/gui.go]: :+1:

**Comment by AudriusButkevicius** [comment]
Otherwise looks good.

**Comment by calmh** [comment]
By the way, at some I may want to refactor the config package... Basically, I'd like to:

 - Let the "load config" machinery also handle the certificates, so the config knows what's our device ID and can handle that appropriately. The config.Config gains MyID() and MyName() and so on so we don't need global variables and/or passing that around separately everywhere.

 - Move some convenience functions from wrapper to the actual config structs, as the wrapper is really mostly supposed to handle saves/loads and the related commit machinery. Then have things that don't need the wrapper (because they're not doing saves or managing subscriptions) keep a config.Config instead of a config.Wrapper for clarity.

And maybe some general tidying...

**Pull request merged**